### PR TITLE
feat: support platformSubFaultDomain node label

### LINKED ISSUE #809: feat: support platformSubFaultDomain as node labels
#### What would you like to be added:

IMDS API version 2021-10-21 would add support platformSubFaultDomain and cloud provider should fetch it and expose as a node label `topology.kubernetes.azure.com/sub-fault-domain`.

IMDS API format:

```
curl "http://169.254.169.254/metadata/instance/compute?api-version=2021-10-01&format=json " -H metadata:true
```

Sample response:

```json
{
    "azEnvironment": "AzurePublicCloud",
    "platformFaultDomain": "5",
    "platformSubFaultDomain": "3",
    "platformUpdateDomain": "4",
    /// ...
}    

```

### PR DESCRIPTION
**Pull request by nilo19**
#### What type of PR is this?

/kind feature

#### Which issue(s) this PR fixes:

Fixes #809

#### Special notes for your reviewer:

This has not been tested because both aks-engine and capz are not supported to create a VMSS with a sub FD.

#### Does this PR introduce a user-facing change?

```release-note
feat: support platformSubFaultDomain node label
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/consts/consts.go [hunk 1]
```diff
@@ -40,9 +40,10 @@ const (
 
 	// LabelFailureDomainBetaZone refer to https://github.com/kubernetes/api/blob/8519c5ea46199d57724725d5b969c5e8e0533692/core/v1/well_known_labels.go#L22-L23
 	LabelFailureDomainBetaZone = "failure-domain.beta.kubernetes.io/zone"
-
 	// LabelFailureDomainBetaRegion failure-domain region label
 	LabelFailureDomainBetaRegion = "failure-domain.beta.kubernetes.io/region"
+	// LabelPlatformSubFaultDomain is the label key of platformSubFaultDomain
+	LabelPlatformSubFaultDomain = "topology.kubernetes.azure.com/sub-fault-domain"
```
Comment (MartinForReal) [inline pkg/consts/consts.go:46]: As documented [here](https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesiozone), I'm wondering if we need to update the prefix of LabelFailureDomainBetaZone and LabelFailureDomainBetaRegion

Comment (feiskyer) [reply pkg/consts/consts.go:46]: yes, we should plan to remove the beta keyword in our labels, but that should be a different topic

**Review by feiskyer — APPROVED** [review]
Thanks for adding this feature

/lgtm
/approve

**Pull request merged**